### PR TITLE
 Refactor ObjCExportHeaderGenerator to allow per declaration translation

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**
Also split ObjCExport header and code generation phases.

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt [hunk 1]
```diff
@@ -275,6 +284,7 @@ internal val toplevelPhase = namedUnitPhase(
         name = "Compiler",
         description = "The whole compilation process",
         lower = frontendPhase then
+                objCExportPhase then
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt]: This is an interesting problem, if the pipeline to be built conditionally for each platform or include noop phases like this one. @alexander-gorshenev WDYT?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt [hunk 2]
```diff
@@ -47,6 +48,14 @@ internal val frontendPhase = konanUnitPhase(
...
 
+internal val objCExportPhase = konanUnitPhase(
+        op = {
+            objCExport = ObjCExport(this)
+        },
+        name = "ObjCExport",
+        description = "Objective-C header generation"
+)
+
 internal val psiToIrPhase = konanUnitPhase(
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt]: And I'd  like top level `object` here.

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt]: How is this related to my PR?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt (hunk 1 above)
Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ToplevelPhases.kt]: Unfortunately, header generation phase is part of frontend. Luckily, this phase doesn't affect IR.
Btw, frontend is already different for different backends.

**Pull request merged**